### PR TITLE
circup: add page

### PR DESCRIPTION
**Pull request by ajs256**
- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

Adds a new page for [circup](https://github.com/adafruit/circup).

### CODE REVIEW

File: pages/common/circup.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+- Search for a library:
+
+`circup show {{partial_name}}`
+
+- List all libraries on a connected device in `requirements.txt` format:
+
+`circup freeze`
+
+- ...and save it to your current directory:
```
Comment (mfrw) [inline pages/common/circup.md:22]:
```suggestion
- Save all libraries on a connected device in current directory:
```

**Comment by askpatrickw** [comment]
Might want to ad the --help output



```
Usage: circup [OPTIONS] COMMAND [ARGS]...

  A tool to manage and update libraries on a CircuitPython device.

Options:
  --verbose         Comprehensive logging is sent to stdout.
  --path DIRECTORY  Path to CircuitPython directory. Overrides automatic path
                    detection.
  --version         Show the version and exit.
  --help            Show this message and exit.

Commands:
  freeze     Output details of all the modules found on the connected...
  install    Install a named module(s) onto the device.
  list       Lists all out of date modules found on the connected...
  show       Show a list of available modules in the bundle.
  uninstall  Uninstall a named module(s) from the connected device.
  update     Update modules on the device. Use --all to automatically update
             all modules without Major Version warnings.
```

**Comment by marchersimon** [comment]
@askpatrickw that's not how tldr pages are structured, but we could add
```
- Show help for `circup`:

`circup --help`
```

File: pages/common/circup.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+- Search for a library:
+
+`circup show {{partial_name}}`
+
+- List all libraries on a connected device in `requirements.txt` format:
+
+`circup freeze`
+
+- Save all libraries on a connected device in current directory:
```
Comment (marchersimon) [inline pages/common/circup.md:22]: Just a minor suggestion:
```suggestion
- Save all libraries on a connected device in the current directory:
```

**Review by navarroaxel — APPROVED** [review]

**Review by marchersimon — APPROVED** [review]
Thanks @ajs256

**Pull request merged**